File: java/src/be/steinsomers/bron_kerbosch/SimpleReporter.java
package be.steinsomers.bron_kerbosch;

import java.util.ArrayList;
import java.util.List;

public final class SimpleReporter implements Reporter {
    public ArrayList<List<Integer>> cliques = new ArrayList<>();

    public void record(List<Integer> clique) {
        assert clique.size() > 1;
        cliques.add(clique);
    }
}

File: csharp/BronKerbosch/SimpleReporter.cs
using System.Collections.Generic;
using System.Collections.Immutable;
using System.Diagnostics;

namespace BronKerbosch
{
    public sealed class SimpleReporter : IReporter
    {
        public List<ImmutableArray<Vertex>> Cliques { get; } = new List<ImmutableArray<Vertex>>();

        public void Record(ImmutableArray<Vertex> clique)
        {
            Debug.Assert(clique.Length > 1);
            Cliques.Add(clique);
        }
    }
}
